scene/gui/code_edit.cpp

```diff
@@ -2379,7 +2379,7 @@ void CodeEdit::confirm_code_completion(bool p_replace) {
 
 		// Handle merging of symbols eg strings, brackets.
 		const String line = get_line(caret_line);
-		char32_t next_char = line[get_caret_column(i)];
+		char32_t next_char = get_caret_column(i) < line.length() ? line[get_caret_column(i)] : 0;
```

Comment:
The caret is always guaranteed to be on the line, the issue is that this doesn't support muli-line insertions. 
Instead we should either remove `\n` from the insert text, or update the caret line before getting the text. `caret_line = get_caret_line(i);`